gdal/port/cpl_aws.cpp

```diff
@@ -600,35 +600,70 @@ static bool Iso8601ToUnixTime(const char* pszDT, GIntBig* pnUnixTime)
 static bool IsMachinePotentiallyEC2Instance()
 {
 #ifdef __linux
-    // Small optimization on Linux. If the kernel is recent
-    // enough, a /sys/hypervisor/uuid file should exist on EC2 instances
-    // and contain a string beginning with ec2.
+    // Optimization on Linux to avoid the network request
     // See http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/identify_ec2_instances.html
-    // If /sys/hypervisor exists, but /sys/hypervisor/uuid does not
-    // exist or doesn't start with ec2, then do not attempt any network
-    // access
-    bool bAttemptNetworkAccess = true;
-    if( CPLTestBool(CPLGetConfigOption(
-                        "CPL_AWS_CHECK_HYPERVISOR_UUID", "YES")) )
+    // Skip if either:
+    // - CPL_AWS_AUTODETECT_EC2=NO
+    // - CPL_AWS_CHECK_HYPERVISOR_UUID=NO (deprecated)
+
+    // On the newer Nitro Hypervisor (C5, M5, H1, T3), use 
+    // /sys/devices/virtual/dmi/id/sys_vendor = 'Amazon EC2' instead.
+
+    // On older Xen hypervisor EC2 instances, a /sys/hypervisor/uuid file will
+    // exist with a string beginning with 'ec2'.
+
+    // If the files exist but don't contain the correct content, then we're not EC2 and
+    // do not attempt any network access
+
+    if( ! CPLTestBool(CPLGetConfigOption("CPL_AWS_AUTODETECT_EC2", "YES")) )
     {
-        VSIStatBufL sStat;
-        if( VSIStatL("/sys/hypervisor", &sStat) == 0 )
+        return true;
+    }
+    else
+    {
+        CPLString opt = CPLGetConfigOption("CPL_AWS_CHECK_HYPERVISOR_UUID", nullptr);
```

Comment:
assigning nullptr to CPLString will crash. Use "" as default value for CPLGetConfigOption() and test for !opt.empty()